internal/bminventory/inventory.go

```diff
@@ -3637,46 +3637,64 @@ func (b *bareMetalInventory) refreshClusterStatus(
+	presigned, err := b.getPresignedForClusterFiles(ctx, &params.ClusterID, params.FileName, params.AdditionalName, params.HostID, params.LogsType)
+	if err != nil {
+		return err
+	}
+
+	return installer.NewV2GetPresignedForClusterFilesOK().WithPayload(presigned)
+}
+
 func (b *bareMetalInventory) GetPresignedForClusterFiles(ctx context.Context, params installer.GetPresignedForClusterFilesParams) middleware.Responder {
+	presigned, err := b.getPresignedForClusterFiles(ctx, &params.ClusterID, params.FileName, params.AdditionalName, params.HostID, params.LogsType)
+	if err != nil {
+		return err
+	}
+
+	return installer.NewGetPresignedForClusterFilesOK().WithPayload(presigned)
+}
+
+func (b *bareMetalInventory) getPresignedForClusterFiles(ctx context.Context, clusterId *strfmt.UUID, fileName string, additionalName *string, hostID *strfmt.UUID, logsType *string) (*models.Presigned, middleware.Responder) {
 	log := logutil.FromContext(ctx, b.log)
 	// Presigned URL only works with AWS S3 because Scality is not exposed
 	if !b.objectHandler.IsAwsS3() {
-		return common.NewApiError(http.StatusBadRequest, errors.New("Failed to generate presigned URL: invalid backend"))
+		return nil, common.NewApiError(http.StatusBadRequest, errors.New("Failed to generate presigned URL: invalid backend"))
 	}
 	var err error
-	fullFileName := fmt.Sprintf("%s/%s", params.ClusterID, params.FileName)
-	downloadFilename := params.FileName
-	if params.FileName == manifests.ManifestFolder {
-		if params.AdditionalName != nil {
-			additionalName := *params.AdditionalName
-			fullFileName = manifests.GetManifestObjectName(params.ClusterID, additionalName)
+	fullFileName := fmt.Sprintf("%s/%s", clusterId, fileName)
+	downloadFilename := fileName
+	if fileName == manifests.ManifestFolder {
+		if additionalName != nil {
+			additionalName := *additionalName
+			fullFileName = manifests.GetManifestObjectName(*clusterId, additionalName)
 			downloadFilename = additionalName[strings.LastIndex(additionalName, "/")+1:]
 		} else {
 			err = errors.New("Additional name must be provided for 'manifests' file name, prefaced with folder name, e.g.: openshift/99-openshift-xyz.yaml")
-			return common.GenerateErrorResponder(err)
+			return nil, common.GenerateErrorResponder(err)
 		}
 	}
 
-	if params.FileName == "logs" {
-		if params.HostID != nil && swag.StringValue(params.LogsType) == "" {
-			logsType := string(models.LogsTypeHost)
-			params.LogsType = &logsType
+	if fileName == "logs" {
+		if hostID != nil && swag.StringValue(logsType) == "" {
+			logsType = swag.String(string(models.LogsTypeHost))
 		}
-		fullFileName, downloadFilename, err = b.getLogFileForDownload(ctx, &params.ClusterID, params.HostID, swag.StringValue(params.LogsType))
+		fullFileName, downloadFilename, err = b.getLogFileForDownload(ctx, clusterId, hostID, swag.StringValue(logsType))
```

Comment:
getLogFileForDownload uses getHost function, which in turn uses GetHostFromDB based on cluster Id and host Id. In V2 it will cause a bug since InfraEnvID will not be equal to ClusterID

Reply:
This API call is for cluster not infra env `/v2/clusters/{cluster_id}/downloads/files-presigned:` is that wrong?

Reply:
no, that's ok, because we are downloading the files of the cluster

Reply:
Ok, IIUC there we can switch to use `common.GetClusterHostFromDB` for both v1/v2 calls right? I switched it.